models/users.js

```diff
@@ -32,6 +32,19 @@ const addOrUpdate = async (userData, userId = null) => {
     if (userId !== null) {
       const user = await userModel.doc(userId).get();
       const isNewUser = !user.data();
+
+      if (Object.keys(userData).includes("member") || Object.keys(userData).includes("archived")) {
+        const roles = { ...user.data().roles };
+        const newRolesArray = Object.entries(userData);
+        if (roles[newRolesArray[0][0]] === newRolesArray[0][1]) return { isRoleUpdated: false };
+        await userModel.doc(userId).update({
+          ...user.data(),
+          roles: { ...user.data().roles, ...userData },
+        });
+
+        return { isRoleUpdated: true };
```

Comment:
Why are we checking this it there
This function should only update it, finding out what needs to be updated should not be coupled with the update function it self

Reply:
Done please check latest commit